code/modules/mining/ore_veins.dm

```diff
@@ -0,0 +1,207 @@
+	icon_state = "geyser"
+	anchored = TRUE
+	layer = LOW_ITEM_LAYER
+	move_resist = INFINITY
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+	var/mining_charges = 6
+	//Classification of the quality of possible ores within a vein, used to determine difficulty
+	var/vein_class = 1
+	//A weighted list of all possible ores that can generate in a vein
+	//The design process is that class 1 veins have a small chance of generating with class 2 ores and so on
+	//As higher class veins will be increasingly harder to mine
+	var/list/ore_list = list(
+		/obj/item/stack/ore/iron = 7,
+		/obj/item/stack/ore/plasma = 3,
+		/obj/item/stack/ore/silver = 2,
+		/obj/item/stack/ore/uranium = 1,
+		/obj/item/stack/ore/titanium = 2,
+		)
+	//The post initialize list of all possible drops from the vein
+	//Meant to be player facing in the form of mining scanners
+	//Contents won't be randomized if the list isn't empty on initialize
+	var/list/vein_contents = list()
+	//Allows subtyped veins to determine how long it takes to mine one mining charge
+	var/mine_time_multiplier = 1
+	//Allows subtyped veins to determine how much loot is dropped per drop_ore call
+	var/drop_rate_amount_min = 15
+	var/drop_rate_amount_max = 20
+	//Mob spawning variables
+	var/spawning_started = FALSE
+	var/max_mobs = 6
+	var/spawn_time = 150 //15 seconds
+	var/mob_types = list(
+		/mob/living/simple_animal/hostile/asteroid/goliath/beast/tendril = 60,
+		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/tendril = 20,
+		/mob/living/simple_animal/hostile/asteroid/brimdemon = 20,
+		)
+	var/spawn_text = "emerges from"
+	var/faction = list("hostile","mining")
+	var/spawn_sound = list('sound/effects/break_stone.ogg')
+	var/spawner_type = /datum/component/spawner
+	var/datum/component/spawner/spawner
+	var/spawn_distance_min = 4
+	var/spawn_distance_max = 6
+	var/wave_length = 2 MINUTES
+	var/wave_downtime = 30 SECONDS
+
+
+//Generates amount of ore able to be pulled from the vein (mining_charges) and types of ore within it (vein_contents)
+/obj/structure/vein/Initialize()
+	. = ..()
+	var/ore_type_amount
+	mining_charges = rand(round(mining_charges - 2),mining_charges + 2)
+	if(!LAZYLEN(vein_contents))
+		switch(vein_class)
+			if(1)
+				ore_type_amount = rand(1,3)
+			if(2)
+				ore_type_amount = rand(3,5)
+			if(3)
+				ore_type_amount = rand(4,6)
+			else
+				ore_type_amount = 1
+		for(ore_type_amount, ore_type_amount>0, ore_type_amount--)
+			var/picked
+			picked = pickweight(ore_list)
+			vein_contents.Add(picked)
+			ore_list.Remove(picked)
+	GLOB.ore_veins += src
+
+/obj/structure/vein/Destroy()
+	. = ..()
+	GLOB.ore_veins -= src
+
+/obj/structure/vein/deconstruct(disassembled)
+	destroy_effect()
+	return..()
+
+/obj/structure/vein/proc/begin_spawning()
+	spawner = AddComponent(spawner_type, mob_types, spawn_time, faction, spawn_text, max_mobs, spawn_sound, spawn_distance_min, spawn_distance_max, wave_length, wave_downtime)
```

Comment:
Usually we don't store components as variables-- anything we need to do interact with it, we do with signals. Might be useful to make a start_ and stop_spawning signal, or something, if you need to.

Just removing the component works too, though